client/state/wp-data/calypso-wp-data-store.js

```diff
@@ -0,0 +1,20 @@
+/** @format */
+
+/**
+ * Internal dependencies
+ */
+import { fetchPreferences, setPreference, savePreference } from 'state/preferences/actions';
```

Comment:
Would be great if we can have our own resolvers, and not rely on any wpcom.undocumented requests.

Looks like we could use `apiFetch()` and register [our own middleware there](https://github.com/WordPress/gutenberg/blob/49c65b9f8c6df42717abb088064a55fb8abbe1ac/packages/api-fetch/src/index.js#L17) for handling WP.com API requests. Has this been considered?

Reply:
Could be arguably deemed a separate concern:

1. Get the adapter to work in current Calypso (this PR)
2. Add middleware to adapt to different APIs (depending on whether we're within Calypso, or producing a bundle that's supposed to run on a self-hosted site)

If we're confident that we _can_ solve the latter given the tools that we have, we can tackle it separately.